dev-util/android-sdk-update-manager/android-sdk-update-manager-26.1.1.ebuild

```diff
@@ -0,0 +1,109 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+PYTHON_COMPAT=(python3_{10..11})
+
+inherit python-r1 udev distutils-r1
+
+DESCRIPTION="Open Handset Alliance's Android SDK"
+HOMEPAGE="https://developer.android.com"
+SRC_URI="https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip"
+RESTRICT="mirror"
+
+LICENSE="android"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="app-arch/tar
+		app-arch/unzip
+		app-arch/gzip
+		acct-group/android"
+
+RDEPEND=">=virtual/jdk-1.5
+	>=dev-java/ant-core-1.6.5
+	dev-java/swt:3.7[cairo]
+	sys-libs/ncurses-compat:5[abi_x86_32(-)]
+	sys-libs/zlib[abi_x86_32(-)]
+"
+
+ANDROID_SDK_DIR="/opt/${PN}"
+QA_FLAGS_IGNORED_x86="
+	${ANDROID_SDK_DIR/\//}/tools/adb
+	${ANDROID_SDK_DIR/\//}/tools/mksdcard
+	${ANDROID_SDK_DIR/\//}/tools/sqlite3
+	${ANDROID_SDK_DIR/\//}/tools/hprof-conv
+	${ANDROID_SDK_DIR/\//}/tools/zipalign
+	${ANDROID_SDK_DIR/\//}/tools/dmtracedump
+"
+QA_FLAGS_IGNORED_amd64="${QA_FLAGS_IGNORED_x86}"
+
+QA_PREBUILT="${ANDROID_SDK_DIR/\//}/tools/*"
+
+S="${WORKDIR}/tools"
+
+src_compile() {
+    :
+}
+
+src_prepare() {
+    rm -rf tools/lib/x86*
+    eapply_user
+}
+
+src_install() {
+    if use udev; then
+        udev_reload
+    fi
+    dodoc "${S}/NOTICE.txt"
+
+    dodir "${ANDROID_SDK_DIR}/tools"
+
+    cp -pPR "${S}"/* "${ED}${ANDROID_SDK_DIR}/tools" || die "failed to install tools"
+    rm -rf "${ED}${ANDROID_SDK_DIR}/tools/lib64" || die "failed to remove lib64 tools"
+
+    # Maybe this is needed for the tools directory too.
+    dodir "${ANDROID_SDK_DIR}"/{add-ons,build-tools,docs,extras,platforms,platform-tools,samples,sources,system-images,temp}
+    fowners -R root:android "${ANDROID_SDK_DIR}"/{.,add-ons,build-tools,docs,extras,platforms,platform-tools,samples,sources,system-images,temp,tools}
+    fperms -R 0775 "${ANDROID_SDK_DIR}"/{.,add-ons,build-tools,docs,extras,platforms,platform-tools,samples,sources,system-images,temp,tools}
+    echo "PATH=\"${EPREFIX}${ANDROID_SDK_DIR}/tools:${EPREFIX}${ANDROID_SDK_DIR}/platform-tools\"" > "${T}/80${PN}" || die
+
+    SWT_PATH=
+    SWT_VERSIONS="3.7 3.6"
+    for version in $SWT_VERSIONS; do
+        # redirecting stderr to /dev/null
+        # not sure if this is best, but avoids misleading error messages
+        SWT_PATH="$(dirname $(java-config -p swt-$version 2> /dev/null) 2> /dev/null)"
+        if [ $SWT_PATH ]; then
+            einfo "SWT_PATH=$SWT_PATH selecting version $version of SWT."
+            break
+        fi
+    done
+
+    echo "ANDROID_SWT=\"${SWT_PATH}\"" >> "${T}/80${PN}" || die
+    echo "ANDROID_HOME=\"${EPREFIX}${ANDROID_SDK_DIR}\"" >> "${T}/80${PN}" || die
+
+    doenvd "${T}/80${PN}"
+
+    echo "SEARCH_DIRS_MASK=\"${EPREFIX}${ANDROID_SDK_DIR}\"" > "${T}/80${PN}" || die
+
+    insinto "/etc/revdep-rebuild" && doins "${T}/80${PN}"
+
+    udev_dorules "${FILESDIR}"/80-android.rules || die
```

Comment:
Same

Reply:
`udev_dorules` does not require `|| die`

https://devmanual.gentoo.org/ebuild-writing/error-handling/#the-die-function
https://devmanual.gentoo.org/eclass-reference/udev.eclass/index.html